python/tk_multi_workfiles/actions/file_action_factory.py

```diff
@@ -36,158 +37,336 @@
-            actions.append(OpenPublishAction(file, file_versions, work_area))
-            if file.path:
-                # file has a local path so it's possible to carry copy it to the local work area! 
-                actions.append(ContinueFromPublishAction(file, current_user_file_versions, work_area))
+            if self._change_work_area and self._can_copy_to_work_area:
+                actions.append(CopyAndOpenPublishInCurrentWorkAreaAction(file_item, file_versions, self._work_area))
+
+        return actions
+
+    def _create_previous_versions_actions_menus(self, file_item):
+        """
+        Creates a list of previous versions menus if the file item has previous versions.
 
-                if change_work_area and can_copy_to_work_area:
-                    actions.append(CopyAndOpenPublishInCurrentWorkAreaAction(file, current_user_file_versions, work_area))
+        :param file_item: File item to generate actions for.
 
-        if not in_other_users_sandbox:
-            if NewFileAction.can_do_new_file(work_area):
-                # New file action
+        :returns: List of actions.
+        """
+        actions = []
+        # ------------------------------------------------------------------
+        actions.append(SeparatorAction())
+
+        actions.extend(
+            self._create_previous_versions_actions_menu(
+                "Previous Work Files",
+                [item for item in file_item.versions.itervalues() if file_item.version > item.version and item.is_local]
+            )
+        )
+
+        actions.extend(
+            self._create_previous_versions_actions_menu(
+                "Previous Publishes",
+                [item for item in file_item.versions.itervalues() if file_item.version > item.version and item.is_published],
+            )
+        )
+
+        return actions
+
+    def _create_previous_versions_actions_menu(self, label, previous_versions):
+        """
+        Creates a menu of actions if the file has previous versions.
+
+        :param label: Name of the menu.
+        :previous_versions: List of versions to generate a menu for.
+
+        :returns: List of menu of actions.
+        """
+        if not previous_versions:
+            return []
+
+        version_versions_actions = self._create_previous_versions_actions(
+            previous_versions
+        )
+        return [ActionGroup(label, version_versions_actions)]
+
+    def _create_previous_versions_actions(self, previous_versions):
+        """
+        Creates a list of actions for all the previous versions of a given type.
+
+        :param pick_locals: True if we want to show previous actions for workfiles, False if we want them for publishes.
+
+        :returns: List of Actions.
+        """
+        previous_versions_actions = []
+
+        # Gives us the last ten versions, from the latest to the earliest.
+        for previous_file_item in previous_versions[-1:-11:-1]:
```

Comment:
maybe make the number of versions to show a variable set at the class level? Would make it slightly easier to change down the road if need be. very minor.